src/main/java/io/hops/hopsworks/expat/migrations/featurestore/featuregroup/UpdateProvIndicesFGDescription.java

```diff
@@ -0,0 +1,254 @@
+          return new Try.Success<>("done");
+        });
+      } else {
+        EpipeRunner.stopEpipe();
+        traverseProvIndices(processIndex("rollback", rollbackScript));
+        EpipeRunner.restartEpipe();
+      }
+    } catch (Throwable e) {
+      throw new RollbackException("error", e);
+    } finally {
+      try {
+        close();
+      } catch (SQLException e) {
+        throw new RollbackException("error", e);
+      }
+    }
+  }
+  
+  private PreparedStatement getProjectInodeStmt(ResultSet allProjectsResultSet) throws SQLException {
+    PreparedStatement projectInodeStmt = connection.prepareStatement(GET_INODE);
+    projectInodeStmt.setLong(GET_INODE_W_PARTITION_ID,
+      allProjectsResultSet.getLong(GET_ALL_PROJECTS_S_PARTITION_ID));
+    projectInodeStmt.setLong(GET_INODE_W_PARENT_ID,
+      allProjectsResultSet.getLong(GET_ALL_PROJECTS_S_INODE_PID));
+    projectInodeStmt.setString(GET_INODE_W_NAME,
+      allProjectsResultSet.getString(GET_ALL_PROJECTS_S_INODE_NAME));
+    return projectInodeStmt;
+  }
+  
+  private void traverseProvIndices(Function<String, Try<String>> provIndexAction) throws Throwable {
+    PreparedStatement allProjectsStmt = null;
+    PreparedStatement projectInodeStmt = null;
+    try {
+      connection.setAutoCommit(false);
+      //get all projects
+      allProjectsStmt = connection.prepareStatement(GET_ALL_PROJECTS);
+      ResultSet allProjectsResultSet = allProjectsStmt.executeQuery();
+      while (allProjectsResultSet.next()) {
+        //get project inode
+        projectInodeStmt = getProjectInodeStmt(allProjectsResultSet);
+        ResultSet projectInodeResultSet = projectInodeStmt.executeQuery();
+        if(!projectInodeResultSet.next()) {
+          throw new IllegalStateException("project inode not found");
+        }
+        Long projectInodeId = projectInodeResultSet.getLong(GET_INODE_S_ID);
+        String provIndex = projectInodeId + "__file_prov";
+        provIndexAction.apply(provIndex).checkedGet();
+      }
+    } finally {
+      if (allProjectsStmt != null) {
+        allProjectsStmt.close();
+      }
+      if(projectInodeStmt != null) {
+        projectInodeStmt.close();
+      }
+    }
+  }
+  
+  private Function<String, Try<String>> processIndex(String type, String script) {
+    return (String indexName) -> {
+      try {
+        LOGGER.info("{} prov index:{}", type, indexName);
+        ElasticClient.reindex(httpClient, elastic, elasticUser, elasticPass, indexName, "temp_" + indexName, script);
+        LOGGER.info("{} prov index:{} restructured mapping", type, indexName);
+        ElasticClient.deleteIndex(httpClient, elastic, elasticUser, elasticPass, indexName);
+        Thread.sleep(2000);
+        ElasticClient.reindex(httpClient, elastic, elasticUser, elasticPass, "temp_" + indexName, indexName);
+        ElasticClient.deleteIndex(httpClient, elastic, elasticUser, elasticPass, "temp_" + indexName);
```

Comment:
I am using temp as an intermediate index, move stuf from index to temp_index(wuth structure changes) and then move stuf from temp_index to index because everyone expects index to contain the actual data